ui/src/dataExplorer/components/SaveAsOverlay.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {PureComponent} from 'react'
+import {withRouter, WithRouterProps} from 'react-router'
+
+// Components
+import SaveAsCellForm from 'src/dataExplorer/components/SaveAsCellForm'
+import SaveAsTaskForm from 'src/dataExplorer/components/SaveAsTaskForm'
+import {Radio, Overlay} from 'src/clockface'
+
+// Styles
+import 'src/dataExplorer/components/SaveAsButton.scss'
+
+enum SaveAsOption {
+  Dashboard = 'dashboard',
+  Task = 'task',
+}
+
+interface State {
+  saveAsOption: SaveAsOption
+}
+
+class SaveAsOverlay extends PureComponent<WithRouterProps, State> {
+  public state: State = {
+    saveAsOption: SaveAsOption.Dashboard,
+  }
+
+  render() {
+    const {saveAsOption} = this.state
+
+    return (
+      <Overlay visible={true}>
+        <Overlay.Container maxWidth={600}>
+          <Overlay.Heading title="Save As" onDismiss={this.handleHideOverlay} />
+          <Overlay.Body>
+            <div className="save-as--options">
+              <Radio>
+                <Radio.Button
+                  active={saveAsOption === SaveAsOption.Dashboard}
+                  value={SaveAsOption.Dashboard}
+                  onClick={this.handleSetSaveAsOption}
+                  data-testid="cell-radio-button"
+                >
+                  Dashboard Cell
+                </Radio.Button>
+                <Radio.Button
+                  active={saveAsOption === SaveAsOption.Task}
+                  value={SaveAsOption.Task}
+                  onClick={this.handleSetSaveAsOption}
+                  data-testid="task-radio-button"
```

Comment:
```suggestion
                  data-testid="task--radio-button"
```

[Using the `BEM` naming style](http://getbem.com/naming/)